.github/workflows/build.yml

```
@@ -130,3 +74,359 @@ jobs:
+    steps:
+      - name: Join build lists
+        uses: actions/github-script@v4
+        id: compile-list
+        with:
+          script: |
+            const coreCoverage = `${{ needs.core-coverage.outputs.core-include }}`;
+            const coreCoverageArray = coreCoverage ? JSON.parse(coreCoverage) : [];
+
+            const boardChanges = `${{ needs.board-changes.outputs.boards-include }}`;
+            const boardChangesArray = boardChanges ? JSON.parse(boardChanges) : [];
+
+            const nightly = `${{ needs.nightly.outputs.nightly-include }}`;
+            const nightlyArray = nightly ? JSON.parse(nightly) : [];
+
+            const combined = [...coreCoverageArray, ...boardChangesArray, ...nightlyArray];
+
+            return [...new Map(combined.map(el => [JSON.stringify(el), el])).values()];
+  core-coverage:
+    if: ${{ needs.get-changed-files.outputs.core-changes == 'true' }}
+    runs-on: ubuntu-latest
+    needs: get-changed-files
+    outputs:
+      core-include: ${{ steps.core-list.outputs.result }} 
+    steps:
+      - uses: actions/github-script@v4
+        id: core-list
+        with:
+          script: |
+            // break out to file
+            const coreCoverage = {
+              boards: [
+                'nice_nano_v2',
+                'nrfmicro_13',
+                'proton_c'
+              ],
+              shields: [
+                'corne_left',
+                'corne_right',
+                'romac',
+                'settings_reset',
+                'tidbit'
+              ],
+              include: [
+                {
+                  "board": "bdn9_rev2"
+                },
+                {
+                  "board": "nice60"
+                },
+                {
+                  "board": "nrf52840_m2",
+                  "shield": "m60"
+                },
+                {
+                  "board": "planck_rev6"
+                },
+                {
+                  "board": "proton_c",
+                  "shield": "clueboard_california"
+                },
+                {
+                  "board": "nice_nano_v2",
+                  "shield": "kyria_left",
+                  "cmake-args": "-DCONFIG_ZMK_DISPLAY=y",
+                },
+                {
+                  "board": "nice_nano_v2",
+                  "shield": "kyria_right",
+                  "cmake-args": "-DCONFIG_ZMK_DISPLAY=y",
+                },
+                {
+                  "board": "nice_nano",
+                  "shield": "romac_plus",
+                  "cmake-args": "-DCONFIG_ZMK_RGB_UNDERGLOW=y -DCONFIG_WS2812_STRIP=y"
+                }
+              ]
+            };
+
+            let include = [];
+            
+            coreCoverage.boards.forEach(b => {
+              coreCoverage.shields.forEach(s => {
+                include.push({
+                  board: b,
+                  shield: s
+                });
+              });
+            });
```

The purist in me says, do it functionally, darn it!

```suggestion
            let include = coreCoverage.boards.flatMap(board => coreCoverage.shields.map(shield => ({ board, shield }));
```

(I may have the JS syntax slightly wrong)